v2-apache/Dockerfile

```diff
@@ -72,50 +108,30 @@ COPY --from=build /usr/local/lib/libfuzzy.so.2.1.0                             /
- && echo 'Include /etc/modsecurity.d/setup.conf' >> /usr/local/apache2/conf/extra/httpd-modsecurity.conf \
- && echo 'Include /etc/modsecurity.d/modsecurity.conf' > /etc/modsecurity.d/setup.conf
-
-RUN if [ "$SETTLS" = "True" ]; then \
-      echo "setting TLS"; \
-      sed -i \
-        -e 's/^#\(Include .*httpd-ssl.conf\)/\1/' \
-        -e 's/^#\(LoadModule .*mod_ssl.so\)/\1/' \
-        -e 's/^#\(LoadModule .*mod_socache_shmcb.so\)/\1/' \
-        conf/httpd.conf; \
-      sed -i \
-        -e 's/^ServerName www\.example\.com:443/ServerName \${SERVERNAME}/' \
-        -e 's/^CustomLog \/proc\/self\/fd\/1 \\/CustomLog		${ACCESSLOG} extended/' \
-        -e 's/"%t %h %{SSL_PROTOCOL}x %{SSL_CIPHER}x \\"%r\\" %b"//' \
-        -e 's/ErrorLog \/proc\/self\/fd\/2/ErrorLog		${ERRORLOG}/' \
-        conf/extra/httpd-ssl.conf; \
-    fi
-
-RUN if [ "$SETPROXY" = "True" ]; then \
-      echo "setting Proxy"; \
-      sed -i \
-        -e 's/^#\(LoadModule .*mod_proxy.so\)/\1/' \
-        -e 's/^#\(LoadModule .*mod_proxy_http.so\)/\1/' \
-        conf/httpd.conf; \
-      echo "<IfModule proxy_module>\nInclude conf/extra/httpd-proxy.conf\n</IfModule>" >> /usr/local/apache2/conf/httpd.conf; \
-      if [ "$SETTLS" = "True" ]; then \
-        echo "<ifModule proxy_module>\nSSLProxyEngine on\nProxyPass / \${PROXYLOCATION}\nProxyPassReverse / \${PROXYLOCATION}\n</ifModule>" > /usr/local/apache2/conf/extra/httpd-proxy.conf; \
-      else \
-        echo "<ifModule proxy_module>\nProxyPass / \${PROXYLOCATION}\nProxyPassReverse / \${PROXYLOCATION}\n</ifModule>" > /usr/local/apache2/conf/extra/httpd-proxy.conf; \
-      fi; \
-    fi
+ && sed -i -E 's|#(LoadModule rewrite_module modules/mod_rewrite.so)|\1|' /usr/local/apache2/conf/httpd.conf \
+ && sed -i -E 's|#(LoadModule proxy_module modules/mod_proxy.so)|\1|' /usr/local/apache2/conf/httpd.conf \
+ && sed -i -E 's|#(LoadModule proxy_http_module modules/mod_proxy_http.so)|\1|' /usr/local/apache2/conf/httpd.conf \
+ && sed -i -E 's|#(LoadModule remoteip_module modules/mod_remoteip.so)|\1|' /usr/local/apache2/conf/httpd.conf \
+ && sed -i -E 's|#(LoadModule socache_shmcb_module modules/mod_socache_shmcb.so)|\1|' /usr/local/apache2/conf/httpd.conf \
+ && sed -i -E 's|#(LoadModule ssl_module modules/mod_ssl.so)|\1|' /usr/local/apache2/conf/httpd.conf \
+ && sed -i -E 's|#(Include conf/extra/httpd-default.conf)|\1|' /usr/local/apache2/conf/httpd.conf \
+ && sed -i -E 's|#(Include conf/extra/httpd-proxy.conf)|\1|' /usr/local/apache2/conf/httpd.conf \
+ && sed -i -E 's|#(Include conf/extra/httpd-ssl.conf)|\1|' /usr/local/apache2/conf/httpd.conf \
+ && sed -i -E 's|#(Include conf/extra/httpd-vhosts.conf)|\1|' /usr/local/apache2/conf/httpd.conf \
+ && echo 'Include conf/extra/httpd-locations.conf' >> /usr/local/apache2/conf/httpd.conf \
+ && echo 'Include conf/extra/httpd-modsecurity.conf' >> /usr/local/apache2/conf/httpd.conf
```

Comment:
Please add the following to the Dockerfile

```
RUN chgrp -R 0 /var/log/ /usr/local/apache2/ && \
    chmod -R g=u /var/log/ /usr/local/apache2/
```

To make it work with non root users (i.e. on OpenShift). See https://docs.openshift.com/container-platform/4.1/openshift_images/create-images.html#images-create-guide-openshift_create-images